src/lib/registration/widgets/checkbox_item.rb

```diff
@@ -0,0 +1,191 @@
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Registration
+  module Widgets
+    # A plain Ruby object in charge to build an item "checkbox" representation to be used by
+    # {MasterDetailSelector} widget in a RichText widget.
+    #
+    # FIXME: give support for disabled items; see #icon
+    class CheckboxItem
+      extend Yast::I18n
+
+      textdomain "registration"
+
+      # Map to icons used in GUI to represent all the known statuses in both scenarios, during
+      # installation (`inst` mode) and in a running system (`normal` mode).
+      #
+      # Available statuses are
+      #
+      #   - `[ ]` not selected
+      #   - `[x]` selected
+      #   - `[a]` auto-selected
+      IMAGES = {
+        "inst:[a]:enabled"    => "auto-selected.svg",
+        "inst:[x]:enabled"    => "inst_checkbox-on.svg",
+        "inst:[ ]:enabled"    => "inst_checkbox-off.svg",
+        "inst:[x]disabled"    => "inst_checkbox-on-disabled.svg",
+        "inst:[ ]:disabled"   => "inst_checkbox-off-disabled.svg",
+        "normal:[a]:enabled"  => "auto-selected.svg",
+        "normal:[x]:enabled"  => "checkbox-on.svg",
+        "normal:[ ]:enabled"  => "checkbox-off.svg",
+        # NOTE: Normal theme has no special images for disabled checkboxes
+        "normal:[x]:disabled" => "checkbox-on.svg",
+        "normal:[ ]:disabled" => "checkbox-off.svg"
+      }.freeze
+      private_constant :IMAGES
+
+      # Path to the icons in the system
+      IMAGES_DIR = "/usr/share/YaST2/theme/current/wizard".freeze
+      private_constant :IMAGES_DIR
+
+      # Help text
+      #
+      # @return [String]
+      def self.help
+        # TRANSLATORS: help text for checked check box
+        checked     = _("Selected")
+        # TRANSLATORS: help text for unchecked check box
+        unchecked   = _("Not selected")
+        # TRANSLATORS: help text for automatically checked check box (it has a different look that a
+        # user selected check box)
+        autochecked = _("Auto selected")
+
+        help_text = []
+        help_text << "<p>"
+        help_text << "#{icon_for(:selected)} = #{checked}<br />"
+        help_text << "#{icon_for(:auto_selected)} = #{autochecked}<br />"
+        help_text << "#{icon_for(:deselected)} = #{unchecked}"
+        help_text << "</p>"
+
+        help_text.join
+      end
+
+      # Returns the icon to be used for an item with given status
+      #
+      # @see #icon
+      #
+      # @return [String]
+      def self.icon_for(status)
```

Comment:
this looks very strange. I expect to get it without instance.

Reply:
Fixed :) See https://github.com/yast/yast-registration/pull/483/commits/7488ee247e0d468c2139aac14675947577c4ef36